packages/common/src/store/pages/chat/sagas.ts

```diff
@@ -355,6 +357,33 @@ function* doFetchPermissions(action: ReturnType<typeof fetchPermissions>) {
   }
 }
 
+function* doSetPermissions(action: ReturnType<typeof setPermissions>) {
+  try {
+    const audiusSdk = yield* getContext('audiusSdk')
+    const sdk = yield* call(audiusSdk)
+    yield* call([sdk.chats, sdk.chats.permit], {
+      permit: action.payload.permissions
+    })
+    const currentUserId = yield* select(getUserId)
+    if (!currentUserId) {
+      throw new Error('currentUserId not found')
+    }
+    yield* put(
+      setPermissionsSucceeded({
+        permissions: {
+          [currentUserId]: {
+            current_user_has_permission: true,
+            permits: action.payload.permissions,
+            user_id: encodeHashId(currentUserId)
+          }
+        }
+      })
+    )
+  } catch (e) {
+    console.error('setPermissionsFailed', e)
```

Comment:
Not sure what the long term pattern is supposed to be here. Should we be firing a sentry event?

Reply:
also what's the user experience for when we fail to set this? Should the controls revert state?

Reply:
Also love seeing the SDK usage!

Reply:
cc @rickyrombo I think the RPC will throw an error if it fails, which will prevent the UI from updating in the first place. Probably should fire a sentry event tho good call.

Reply:
Great callout. I haven't been good about explicitly throwing sentry errors and have instead been relying on our catch-all sentry rule of console.errors but I shouldn't do that - I'll make a note to clean some of these up